src/t_string.c

```diff
@@ -380,6 +380,27 @@ void psetexCommand(client *c) {
     setGenericCommand(c, OBJ_PX | OBJ_ARGV3, c->argv[1], c->argv[3], c->argv[2], UNIT_MILLISECONDS, NULL, NULL, NULL);
 }
 
+void delifeqCommand(client *c) {
+    robj *existing_value = lookupKeyWrite(c->db, c->argv[1]);
+
```

Comment:
We should fail with a WRONGTYPE error if someone tries to use this command on a key that's not a string.

Let's add a test case for this too.

```suggestion
    if (checkType(c, existing_value, OBJ_STRING)) return;

```